### PR TITLE
NXopt workshop updates

### PR DESCRIPTION
**Pull request by RonHildebrandt**
This PR contains the updates of NXopt (now renamed to NXoptical_spectroscopy) from the optical spectroscopy workshop in Berlin. Most changes are located in the NXoptical_spectroscopy definition. 

Major adjustments: NXoptical_spectroscopy
- Included reference frame description,
- updated exists/recommended requirements, 
- restructruring the order in the instrument level.

Minor adjustments:
- Adjustments for NXopt_window, NXlens_opt, NXwaveplate were things, which were originally located in NXopt but are of general relevance.
- NXellipsometry had some minor adjustments for the the "exists:" fields.
- NXraman contained only the NXopt renaming and minor "exists:" adjustments.
- For the creation of the manual, the ellipsometry-structure.rst was adjusted, so it contains as well the NXopt renaming.

With this adjustmenets, NXoptical_spectroscopy should be finalized.

### CODE REVIEW

**Comment by tomio13** [comment]
I am looking at the [NXoptical_spectroscopy](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/NXopt_workshop_updates/contributed_definitions/nyaml/NXoptical_spectroscopy.yaml).
On line 182, why is the beam_TYPE half capitalized?

Line 201: incident_wavelength:
From the usage of spread in the consecutive lines, I would assume this is the central wavelength of a spectral range. Please specify this in the doc string of this field.

In Line 207: what does extent stand for? Spatial size? Temporal duration?

In line 229 the docstring is incomplete.

In line 290: "Specify respective hardware which was used to for the detector. "... should be probable 'Specify respective hardware which was used for the detector. '

... for now

**Comment by RonHildebrandt** [comment]
@tomio13 
Line 201 and Line 207: I did not specify this as, I wanted to refer to NIAC definitions: https://manual.nexusformat.org/classes/base_classes/NXbeam.html
In principle, these should be transfered as well to the html site (https://fairmat-nfdi.github.io/nexus_definitions/NXopt_workshop_updates/classes/contributed_definitions/NXoptical_spectroscopy.html#nxoptical-spectroscopy-entry-instrument-lens-opt-type-field). But this is not the case right now.

 Line 229: Finished the docstring.
 
 Line 290: Done.

File: manual/source/classes/contributed_definitions/ellipsometry-structure.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ In the application definition we provide a minimum set of description elements a
 Application Definitions
 -----------------------
 
-    :ref:`NXopt`:
+    :ref:`NXoptical_spectroscopy`:
        A generic application definition for optial spectorscopy measurements, including complex systems up to variable angle spectroscopic ellipsometry. 
 
     :ref:`NXellipsometry`:
```
Comment (sanbrock) [inline manual/source/classes/contributed_definitions/ellipsometry-structure.rst:30]: also add NXraman here

**Review by sanbrock — CHANGES_REQUESTED** [review]
some suggestions

File: contributed_definitions/nyaml/NXwaveplate.yaml [hunk 2]
```diff
@@ -40,6 +40,10 @@ NXwaveplate(NXobject):
     dimensions:
       rank: 1
       dim: [[1, N_wavelengths]]
+  retardance_distribution(NXdata):
+    exists: optional
```
[the remark this answers is not in the document]
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXwaveplate.yaml:44]: done

File: manual/source/classes/contributed_definitions/ellipsometry-structure.rst (hunk 1 above)
Comment (RonHildebrandt) [reply manual/source/classes/contributed_definitions/ellipsometry-structure.rst:43]: I've added Raman and updated the respective documentation with the lastest status of optical spectroscopy.

**Comment by RonHildebrandt** [comment]
@sanbrock I adressed everything. Hope everything is fine now to merge this in.

**Comment by domna** [comment]
@RonHildebrandt Could you change the `definition/@url` to `definition/@URL`? Then it would match the concept `NXentry/definition/@URL` and would be in alignment with https://github.com/FAIRmat-NFDI/pynxtools/pull/361

**Comment by RonHildebrandt** [comment]
@domna added the requested changes

**Review by sanbrock — APPROVED** [review]

**Pull request merged**